File: Code/mailer/SimKnx.php
<?php


class SimKnx {
	
	private $simKnx_host;
	private $simKnx_port;

	/**
	 * Constructor
	 *
	 * @param      string  $host   The SIM_KNX host
	 * @param      string  $port   The SIM_KNX port
	 */
	public function __construct($host = "localhost", $port = 1234)
	{
		$this->simKnx_host = $host;
		$this->simKnx_port = $port;
	}

	/**
	 * Sets a bool object.
	 *
	 * @param      string  $objectNumber  The object number
	 * @param      Bool  $bool          The bool
	 */
	public function setBoolObject($objectNumber, $bool)
	{
		$request = "set " . $objectNumber . " " . $bool;
		$this->setObject($request);
	}

	/**
	 * Sets a dimmer object.
	 *
	 * @param      string	$objectNumber  The object number
	 * @param      Bool    	$direction     The direction
	 * @param      Int 	   	$value         The value
	 */
	public function setDimmerObject($objectNumber, $direction, $value)
	{
		$request = "set " . $objectNumber . " " . $direction . " " . $value;
		$this->setObject($request);
	}

	/**
	 * Sets a integer object.
	 *
	 * @param      string  	$objectNumber  The object number
	 * @param      Int  	$value         The value
	 */
	public function setIntegerObject($objectNumber, $value)
	{
		$request = "set " . $objectNumber . " " . $value;
		$this->setObject($request);
	}

	/**
	 * Send a set request over UDP
	 *
	 * @param      String  $request  The request string preformatted
	 */
	private function setObject($request)
	{
		// Create socket
		$socket = socket_create(AF_INET, SOCK_DGRAM, SOL_UDP);

		// Send the message
		socket_sendto($socket, $request, strlen($request), 0, $this->simKnx_host ,$this->simKnx_port);

		socket_close($socket);
	}

	/**
	 * Get the type of an object
	 *
	 * @param      int  $objectNumber  The object number
	 *
	 * @return     string  The object type.
	 */
	public function getObjectType($objectNumber)
	{

		$message = "type " . $objectNumber;

		// Create socket and bind it to localhost for a response
		$socket = socket_create(AF_INET, SOCK_DGRAM, SOL_UDP);
		socket_bind($socket, "localhost");

		// Send the message and ask for the object type
		socket_sendto($socket, $message, strlen($message), 0, $this->simKnx_host ,$this->simKnx_port);

		$addr = "";
		$port = 0;
		$response = "";

		socket_recvfrom($socket, $response, 10, 0, $addr, $port);

		socket_close($socket);

		return $response;

	}

	/**
	 * Get the value for a object
	 *
	 * @param      Int  $objectNumber  The object number
	 *
	 * @return     string  The object.
	 */
	public function getObject($objectNumber)
	{
		$message = "get " . $objectNumber;

		// Create socket and bind it to localhost for a response
		$socket = socket_create(AF_INET, SOCK_DGRAM, SOL_UDP);
		socket_bind($socket, "localhost");

		// Send the message and ask for the object type
		socket_sendto($socket, $message, strlen($message), 0, $this->simKnx_host ,$this->simKnx_port);

		$addr = "";
		$port = 0;
		$response = "";

		socket_recvfrom($socket, $response, 10, 0, $addr, $port);

		socket_close($socket);

		return $response;
	}
}

File: Code/weather/SimKnx.py
import socket


class SimKnx:

    def __init__(self, host="localhost", port=1234):
        """ Constructor with default parameters """
        self.__SimKnx_host = host
        self.__SimKnx_port = port

    def setBoolObject(self, objectNumber, value: bool):
        """ Sets a value to a boolean object """
        request = 'set {} {}'.format(objectNumber, int(value))
        self.__setObject(request)

    def setDimmerObject(self, objectNumber, direction: bool, value: int):
        """ Sets a dimmer object """
        request = 'set {} {} {}'.format(objectNumber, int(direction), value)
        self.__setObject(request)

    def setintegerObject(self, objectNumber, value: int):
        """ Sets a value to an integer object """
        request = 'set {} {}'.format(objectNumber, value)
        self.__setObject(request)

    def __setObject(self, request):
        """ Send UDP request to set an object """
        # Create UDP socket
        sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)

        # Send the message
        sock.sendto(request.encode("utf-8"), (self.__SimKnx_host, self.__SimKnx_port))

        sock.close()

    def getObjectType(self, objectNumber: int):
        """ Get the type for a SIM-KNX object """
        request = "type {}".format(objectNumber)
        return int(self.__getResponse(request))

    def getObject(self, objectNumber: int):
        """ Get the value for the specified object """
        request = "get {}".format(objectNumber)
        return int(self.__getResponse(request))

    def __getResponse(self, request):
        """ Send the request and get the response back"""

        # Create socket and bind it to localhost for a response
        sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
        sock.bind(('', 4321))

        # Send the message and ask for the object type
        sock.sendto(request.encode("utf-8"), (self.__SimKnx_host, self.__SimKnx_port))
        data = sock.recvfrom(10)
        sock.close()

        return data[0]
